channeldb/invoices.go

```diff
@@ -1690,6 +1758,22 @@ func (d *DB) updateInvoice(hash lntypes.Hash, invoices, settleIndex kvdb.RwBucke
 			return nil, errors.New("nil custom records map")
 		}
 
+		// If a newly added HTLC has an associated set id, use it to
+		// index this invoice in the set id index. An error is returned
+		// if we find the index already points to a different invoice.
+		if htlcUpdate.AMP != nil {
+			setID := htlcUpdate.AMP.SetID()
+			setIDInvNum := setIDIndex.Get(setID[:])
+			if setIDInvNum == nil {
+				err = setIDIndex.Put(setID[:], invoiceNum)
```

Comment:
Does this commit need to clean up logic as well (delete invoice)?

Reply:
We don't support deleting invoices (yet).

Reply:
Oh that is right. I was thinking of `DeleteAllPayments`, but that are payments.